### PR TITLE
Add Create Offline Configuration

### LINKED ISSUE #1957: Ability to use `Flutter: New Project` offline
Hello team.

I'm currently struggling to configure the extension to prefer using the cache over hitting the network due to a spotty connection.

### Current workflow:
Creating a project with the `New Project` command keeps retrying `pub get` indefinitely. Due to unresolved dependencies, the UI hangs until the user cancels. The last time I checked, it retried 236 times before I quit. Running `pub get` via the `Get Packages` command produces the same results.

### Expectation:
Hit the pub registry, but resolve to cache if/when packages have already been cached as per CLI or 
Exit after x number of retries

### Current workflow:
Cancelling the running command seems to only dismiss the toast as it still attempts to get packages.

### Expectation:
Cancel the running command as the action implies or 
Check cache then quit if not resolved

Initially thought `"dart.pubAdditionalArgs": [ "--offline" ]` or `"dart.flutterAdditionalArgs": [ "--offline" ]` would produced the desired outcome but that is not the case. The settings are saved but do not seem to do anything. A setting similar to [this previous suggestion](https://github.com/Dart-Code/Dart-Code/issues/618#issuecomment-368351176) is an ideal method for those who want to configure offline usage. Either that or passing the `--offline` flag to `dart.pubAdditionalArgs` / `dart.flutterAdditionalArgs`, but I prefer the latter as they are already implemented settings.

PS: Although the [`dart.flutterAdditionalArgs` setting](https://dartcode.org/docs/settings/#dartflutteradditionalargs) is documented, VS Code says it's an unknown configuration setting. 

Cheers!

### PR DESCRIPTION
**Pull request by ejabu**
Fixes #1957.

### CODE REVIEW

**Comment by ejabu** [comment]
Issues : 

https://github.com/Dart-Code/Dart-Code/issues/1957

File: src/extension/commands/sdk.ts [hunk 1]
```diff
@@ -210,6 +213,9 @@ export class SdkCommands {
 			if (config.flutterAndroidX) {
 				args.push("--androidx");
 			}
+			if (config.flutterCreateOffline) {
+				args.push("--offline");
+			}
```
Comment (DanTup) [inline src/extension/commands/sdk.ts]: It looks like this argument might be being added twice?

File: src/extension/config.ts [hunk 2]
```diff
@@ -67,6 +67,7 @@ class Config {
 	get flutterAndroidX(): boolean { return this.getConfig<boolean>("flutterAndroidX", false); }
 	get flutterCreateAndroidLanguage(): "java" | "kotlin" { return this.getConfig<"java" | "kotlin">("flutterCreateAndroidLanguage", "kotlin"); }
 	get flutterCreateIOSLanguage(): "objc" | "swift" { return this.getConfig<"objc" | "swift">("flutterCreateIOSLanguage", "swift"); }
+	get flutterCreateOffline(): boolean { return this.getConfig<boolean>("flutterCreateOffline", true); }
```
Comment (DanTup) [inline src/extension/config.ts]: This `true` should be `false` (it should match the default value).

File: package.json [hunk 3]
```diff
@@ -1231,6 +1231,12 @@
 					"description": "Whether to automatically send a Hot Restart request during a debug session when saving files if Hot Reload is not available but Hot Restart is.",
 					"scope": "window"
 				},
+				"dart.flutterCreateOffline": {
+					"type": "boolean",
+					"default": false,
+					"description": "Whether to create new flutter project using offline mode",
```
Comment (DanTup) [inline package.json]: Nit: Should end with full stop to match the others.

**Comment by DanTup** [comment]
I'm planning on a beta release shortly, so I'll merge this and tidy up the small things mentioned above. Thanks!

**Comment by ejabu** [comment]
> I'm planning on a beta release shortly, so I'll merge this and tidy up the small things mentioned above. Thanks!

sorry got no change of working it, kinda busy setting up new laptop.

thanks by the way for merging it !

**Comment by DanTup** [comment]
No worries - I merged it and made those tweaks. It's in the beta I released today:

https://github.com/Dart-Code/Dart-Code/releases/tag/v3.9.0-beta.1

And will ship in the stable release (aiming for end of this week). Thanks for the contribution!

**Pull request merged**